### PR TITLE
get and show data users list

### PR DESCRIPTION
**Pull request by at-trampham**
## Description
List off Users,Can see No., Employee code, Name, email, number of books user donate, number of books user borrowed.
## Reference Requirements
- Product Backlog ID:#PBI-10
- Task ID:  #10.2

## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
select count(*) as aggregate from `users` where `users`.`deleted_at` is null
image
select `users`.`id`, `users`.`employ_code`, `users`.`name`, `users`.`team`, `users`.`email`, `users`.`is_admin`, `users`.`avatar_url`, (select count(*) from `books` where `users`.`employ_code` = `books`.`from_person` and `books`.`deleted_at` is null) as `books_count`, (select count(*) from `borrows` where `users`.`id` = `borrows`.`user_id`) as `borrows_count` from `users` where `users`.`deleted_at` is null order by `id` asc limit 10 offset 0
image
select * from `users` where `id` = '11' and `users`.`deleted_at` is null limit 1
image

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: resources/views/backend/users/index.blade.php [hunk 1]
```diff
@@ -37,53 +37,25 @@
...
-                    <li class="page-item">
-                      <a class="page-link" href="#">Next</a>
-                    </li>
-                  </ul>
+                    <ul class="pagination">
+                        {{ $users->links() }}
+                    </ul>
                 </nav>
               </div>
               <!-- /.pagination -->
```
Comment (at-duongtruong) [inline resources/views/backend/users/index.blade.php]: GLTM

File: app/Http/Controllers/Admin/UserController.php [hunk 2]
```diff
@@ -14,7 +16,23 @@ class UserController extends Controller
...
-        return view('backend.users.index');
+        $fields = [
+            'users.id',
+            'users.employ_code',
+            'users.name',
+            'users.team',
+            'users.email',
+            'users.is_admin',
+            'users.avatar_url',
+            DB::raw('COUNT(DISTINCT(borrows.id)) AS total_borrowed'),
```
Comment (at-tuanphan) [inline app/Http/Controllers/Admin/UserController.php]: em thử câu quẻy này thử:

> $fields = [
            'users.id',
            'users.employ_code',
            'users.name',
            'users.team',
            'users.email',
            'users.is_admin',
            'users.avatar_url',
        ];
        $users = User::select($fields)
        ->withCount(['books', 'borrows'])
        ->orderBy('id')
        ->paginate(config('define.users.row_count'));

Update cái quan hệ `books` trong model `users`

> public function books()
    {
        return $this->hasMany(Book::class, 'from_person', 'employ_code');
    }

File: resources/views/backend/users/index.blade.php [hunk 3]
```diff
@@ -37,53 +37,25 @@
...
-                  <td>Tram Pham T.M.</td>
-                  <td>tram.pham@asiantech.com</td>
-                  <td>2</td>
-                  <td>CSS</td>
+                  <td>{{ $index + $users->firstItem() }}</td>
+                  <td>{{ $user->employ_code }}</td>
+                  <td>{{ $user->name }}</td>
+                  <td>{{ $user->email }}</td>
+                  <td>{{ $user->total_donated }}</td>
+                  <td>{{ $user->total_borrowed }}</td>
```
Comment (at-tuanphan) [inline resources/views/backend/users/index.blade.php]: update lại lúc show total ra như vậy:

> <td>{{ $user->books_count }}</td>
> <td>{{ $user->borrows_count }}</td>

**Review by ghost — COMMENTED** [review]
please check my comments

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**